external/storm-redis/src/main/java/org/apache/storm/redis/state/RedisKeyValueState.java

```diff
@@ -60,22 +64,23 @@ public RedisKeyValueState(String namespace) {
     }
 
     public RedisKeyValueState(String namespace, JedisPoolConfig poolConfig) {
-        this(namespace, poolConfig, new DefaultStateSerializer<K>(), new DefaultStateSerializer<V>());
+        this(namespace, poolConfig, new DefaultStateSerializer<K>(), new DefaultStateSerializer<Optional<V>>());
     }
 
-    public RedisKeyValueState(String namespace, JedisPoolConfig poolConfig, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
+    public RedisKeyValueState(String namespace, JedisPoolConfig poolConfig, Serializer<K> keySerializer, Serializer<Optional<V>> valueSerializer) {
```

Comment:
Sounds good.  If I understood correctly, the user provided serializer would be used to serialize the value to bytes which is then wrapped in an Optional<byte[]> and serialized via the DefaultStateSerializer.